libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -679,6 +696,7 @@ bool WaveClip::Paste(double t0, const WaveClip &other)
          copy->ConvertToSampleFormat(GetSampleFormats().Stored());
       newClip = std::move(copy);
    }
+   newClip->ApplyStretchRatio();
```

Comment:
You apply stretch ratio after any needed resampling or change of format.

Resampling to a lesser rate might be theoretically lossless for a band-limited signal, but maybe not practically.  Change of format to a lesser bit width mixes in some dither noise.

Maybe it's better to stretch first and make those other transformations after, for better quality?

Reply:
Thanks for that remark.
Resampling:
Computational cost: the fastest would be to time-stretch before in case of upsampling, and after in case of downsampling.
Quality: I do not know the algorithm sufficiently to say what influence the order has. I would have guessed that it can only be better with more frequency information, hence in case of downsampling I'd have first stretched and then downsampled. David Hearn doesn't believe this to have a noticeable consequence.

Sample-format:
Computational cost: I suppose calculations are faster on lower precision samples, but I'm largely ignorant on the matter - that'd be a question to @crsib.
Quality: I don't expect a perceivable difference here. If someone is of another opinion I'd consider testing.

The only obvious impact I see is wrt performance when resampling. I'll push something that handles this.

Reply:
Thanks for that remark.
Resampling:
Computational cost: the fastest would be to time-stretch before in case of upsampling, and after in case of downsampling.
Quality: I do not know the algorithm sufficiently to say what influence the order has. I would have guessed that it can only be better with more frequency information, hence in case of downsampling I'd have first stretched and then downsampled. David Hearn doesn't believe this to have a noticeable consequence.

Sample-format:
Computational cost: I suppose calculations are faster on lower precision samples, but I'm largely ignorant on the matter - that'd be a question to @crsib.
Quality: I don't expect a perceivable difference here. If someone is of another opinion I'd consider testing.

The only obvious impact I see is wrt performance when resampling. I'll push something that handles this.